extensions/amp-story/1.0/animation.js

```diff
@@ -74,52 +76,66 @@ const PlaybackActivity = {
   FINISH: 1,
 };
 
+/**
+ * @param {!Element} root
+ * @param {!Element} element
+ * @return {?string}
+ */
+function getSequencingStartAfterId(root, element) {
+  if (!element.hasAttribute(ANIMATE_IN_AFTER_ATTRIBUTE_NAME)) {
+    return null;
+  }
+  const dependencyId = element.getAttribute(ANIMATE_IN_AFTER_ATTRIBUTE_NAME);
+
+  user().assertElement(
+    root.querySelector(`#${escapeCssSelectorIdent(dependencyId)}`),
+    `The attribute '${ANIMATE_IN_AFTER_ATTRIBUTE_NAME}' in tag ` +
+      `'${element.tagName}' is set to the invalid value ` +
+      `'${dependencyId}'. No children of parenting 'amp-story-page' ` +
+      `exist with id ${dependencyId}.`
+  );
+
+  return dependencyId;
+}
+
 /** Wraps WebAnimationRunner for story page elements. */
 export class AnimationRunner {
   /**
    * @param {!Element} page
-   * @param {!WebAnimationDef|!StoryAnimationDef} animationDef
+   * @param {!StoryAnimationConfigDef} config
    * @param {!Promise<!../../amp-animation/0.1/web-animations.Builder>} webAnimationBuilderPromise
    * @param {!../../../src/service/vsync-impl.Vsync} vsync
    * @param {!AnimationSequence} sequence
-   * @param {!Object<string, *>=} keyframeOptions
    */
-  constructor(
-    page,
-    animationDef,
-    webAnimationBuilderPromise,
-    vsync,
-    sequence,
-    keyframeOptions = {}
-  ) {
+  constructor(page, config, webAnimationBuilderPromise, vsync, sequence) {
+    const {source, preset, startAfterId, spec} = config;
+
     /** @private @const */
     this.page_ = page;
 
+    /** @private @const */
+    this.source_ = source;
+
     /** @private @const */
     this.vsync_ = vsync;
 
     /** @private @const {?Element} */
-    this.presetTarget_ = !!animationDef.preset
-      ? dev().assertElement(animationDef.target)
-      : null;
+    this.presetTarget_ = !!preset ? dev().assertElement(spec.target) : null;
 
     /** @private @const */
     this.sequence_ = sequence;
 
-    /** @private @const {string|undefined} */
-    this.startAfterId_ = animationDef.startAfterId;
+    /** @private @const {?string} */
+    this.startAfterId_ = startAfterId;
 
     /** @private @const {!Promise<!WebAnimationDef>} */
-    this.webAnimationDefPromise_ = this.getWebAnimationDef_(
-      animationDef,
-      keyframeOptions
-    );
+    this.resolvedSpecPromise_ = this.resolveSpec_(config);
```

Comment:
Curious about the renaming, maybe it's just me but `resolveSpec` is an extremely generic name that doesn't give me any information about what it does or returns, when `getWebAnimationDef_` was pretty clear?

Reply:
`spec` is the internal name for this type of configuration inside `amp-animation`. I felt like otherwise the two different types of objects could be confused. It's also for uniformity with the new `spec` field.